src/Handler/MockHandler.php

```diff
@@ -87,6 +87,19 @@ function ($value) use ($request, $options) {
                 if ($this->onFulfilled) {
                     call_user_func($this->onFulfilled, $value);
                 }
+                if (isset($options['sink'])) {
+                    $contents = (string) $value->getBody();
+                    $sink = $options['sink'];
+
+                    if (is_resource($sink)) {
```

Comment:
The way I'm using it is "create a tmpfile" pass it as a "sink" parameter (its a smallish zip file that I want downloaded and read, but to make sure it gets deleted afterwards). 
I use get_stream_meta() to get the location of the actual file in the system and then use the php's Zip reading lib to do the job. 
This all works fine, but I couldn't test it out with a mocked request. And implementing it with `stream_for` closes the stream before I could pass it to zip library, giving me a bunch of errors. The way shown in this pull request works fine. 
You can test it out yourself, as the test here is basically doing the same.